src/simulation.jl

```diff
@@ -1,38 +1,47 @@
-using Base.Iterators
 using CombinatorialSpaces
 using ComponentArrays
 using LinearAlgebra
 using MLStyle
 using PreallocationTools
 
-const gensim_in_place_stub = Symbol("GenSim-M")
+const GENSIM_INPLACE_STUB = Symbol("GenSim-M")
+const NO_STUB_RETURN = Symbol("NOSTUB")
 
 abstract type GenerationTarget end
```

Comment:
`AbstractGenerationTarget` would be a quick rename

Reply:
Yeah, I'm realizing now all of our targets should be abstract types to allow for easy extension but at this point this would be breaking so this can be changed later.